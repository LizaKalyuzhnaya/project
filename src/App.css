#root {
  display: flex;
}
.app {
  display: flex;
  justify-content: space-between;
}
.app__search {
  width: 25vw;
  margin-right: 10px;
  max-height: 97vh;
  overflow: auto; 
  box-sizing: border-box;
}
.app__search__form {
  margin: 10px 3px;
  display: flex;
  align-content: center;
  justify-content: center;
  width: 96%;
  box-sizing: border-box;
}
.app__search__form__input {
  height: 40px;
  outline: none;
  border: none;
  width: 80%;
  box-shadow: 0 0 4px rgba(0,0,0, .3);
}
.app__search__form__input:focus {
  box-shadow: 0 0 4px rgba(0, 122, 255, .7);
}
.app__search__form__button {
  width: 20%;
  height: 45px;
  margin-top: -2px;
  background: #758deb;
  color: beige;
  border: none;
}
.app__search__list-place__item {
  list-style-type: none;
  min-height: 40px;
  width: 96%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 10px;
  box-shadow: 0 0 6px rgba(0,0,0, .1);
  margin-top: 10px;
  box-sizing: border-box;
  /* border-top: 1px solid rgba(0,0,0,0.3);
  border-bottom: 1px solid rgba(0,0,0,0.3); */
}
.app__search__list-place__item__text {
  width: 80%;
  word-wrap: break-word;
  text-align: left;
}
.app__search__list-place__item__svg {
  padding-top: 5px;
}
.app__search__list-place  {
  padding: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 96%;
  margin: 0 3px;
}

@media screen and (max-width: 740px) {
  .app__search {
    margin-right: 5px;
  }
  .app__search__form__button {
    display: none;
  }
  .app__search__form__input {
    width: 97%;
    box-sizing: border-box;
  }
}

@media screen and (min-width: 1024px) {
  .app__search {
    width: 27.5vw;
  }
}