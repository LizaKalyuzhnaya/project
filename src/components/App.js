import React, { Component } from 'react';
import Map from '../containers/Map';
import AddPlaceName from '../containers/AddPlaceName';
import List from '../containers/List';
import '../App.css';

class App extends Component {
  render() {
    return (
      <div className = "app">
        <div className = "app__search" >
          <AddPlaceName />
          <List />
        </div>
        <Map/>
      </div>
    );
  }
}

export default App;
