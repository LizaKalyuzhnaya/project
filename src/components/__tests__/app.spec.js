import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from '../../reducers'
import { shallow, mount } from 'enzyme'
import App from '../App'
import myMap from '../Map'

describe('App', () => {
    describe('App mount', () => {
        const store = createStore(rootReducer);
        const myComponent = mount(
                                <Provider store={store}>
                                    <App />
                                </Provider>
                                );
                                
        it('render initial', () => {
            expect(myComponent.find('AddPlaceName')).toHaveLength(1);
            expect(myComponent.find('List')).toHaveLength(1);
            expect(myComponent.find(myMap)).toHaveLength(1);
        })
    })

    describe('App shallow', () => {
        const myComponent = shallow(<App />);

        it('renders properly', () => {
            expect(myComponent).toMatchSnapshot()
        })
    })    
})