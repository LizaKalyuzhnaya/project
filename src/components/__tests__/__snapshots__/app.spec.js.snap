// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App shallow renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="app__search"
        >
          <Connect(AddPlaceName) />
          <Connect(List) />
        </div>,
        <Connect(withProps(withHandlers(withScriptjs(withGoogleMap(Component))))) />,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(AddPlaceName) />,
            <Connect(List) />,
          ],
          "className": "app__search",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="app__search"
          >
            <Connect(AddPlaceName) />
            <Connect(List) />
          </div>,
          <Connect(withProps(withHandlers(withScriptjs(withGoogleMap(Component))))) />,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(AddPlaceName) />,
              <Connect(List) />,
            ],
            "className": "app__search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
